samcli/commands/check/resources/graph.py

```diff
@@ -0,0 +1,53 @@
+"""
+Class for graph. All data is stored in the graph directly, or within nodes that are stored in the graph
+"""
+from typing import List, Optional, Union
+
+from samcli.commands.check.resources.lambda_function_pricing import LambdaFunctionPricing
+from samcli.commands.check.resources.lambda_function import LambdaFunction
+from samcli.commands.check.resources.warning import CheckWarning
+
+
+class CheckGraph:
+    """
+    A graph object stores four main item types.
+
+    entry_points: All possible entry points to the application based on the template.
+    resources_to_analyze: All resources that need to be directly checked for bottle
+      neck issues (just lambda functions for now).
+    green/yellow/red/red_burst_warnings: The four different warning types that
+      the user can be presented
+    lambda_function_pricing_info: Contains the user entered data for the lambda
+      pricing questions
+    """
+
+    entry_points: List[Union[LambdaFunction]]
+    resources_to_analyze: List[Union[LambdaFunction]]
+    green_warnings: List[CheckWarning]
+    yellow_warnings: List[CheckWarning]
+    red_warnings: List[CheckWarning]
+    red_burst_warnings: List[CheckWarning]
+    lambda_function_pricing_info: Optional[LambdaFunctionPricing]
+
+    def __init__(self, lambda_functions: List):
```

Comment:
Can we be more specific about the List like `List[LambdaFunction]`?

Reply:
Should this line be `def __init__(self, lambda_function: List[LambdaFunction])`?